### PR TITLE
Add basic grades logic.

### PR DESCRIPTION
**Pull request by Jonas-Sander**
Add `GradesService` and implement adding a Term with it in a basic manner.

Different kind of grades and more will be added in a future PRs.

In this PR:
* I added a `GradesService` with tests
* I added a "Neues Halbjahr hinzufügen" tile
* I added a very rough create term dialog
* Implement adding a term by using `GradesService`
* Test adding a term in `grades_widget_test.dart`

image

image

image

Code is not perfect, I have more classes than necessary currently. I'll further refactor in future PRs.

### CODE REVIEW

**Comment by Jonas-Sander** [comment]
@nilsreichardt Could you update the goldens? Since you didn't add a button this is just temporary until you implement adding a term.

File: app/lib/navigation/drawer/tiles/drawer_tiles.dart [hunk 1]
```diff
@@ -22,6 +22,7 @@ const onlyDesktopTiles = <Widget>[
   DrawerTile(NavigationItem.homework),
   DrawerTile(NavigationItem.timetable),
   DrawerTile(NavigationItem.blackboard),
+  DrawerTile(NavigationItem.grades),
```
Comment (nilsreichardt) [inline app/lib/navigation/drawer/tiles/drawer_tiles.dart:25]: If you merge the `main` into this branch, we can access the grades page with `Grades` as activation code.

Comment (nilsreichardt) [inline app/lib/navigation/drawer/tiles/drawer_tiles.dart:25]:
```suggestion
```

If you merge the `main` into this branch, we can access the grades page with `Grades` as activation code.

**Comment by Jonas-Sander** [comment]
@nilsreichardt is it okay like this? If so please approve the PR

**Review by nilsreichardt — APPROVED** [review]
LGTM

Sorry, forgot to send the review

**Pull request merged**